### PR TITLE
Problem: missing logic to parse MsgExec internal messages

### LINKED ISSUE #525: Problem: missing logic to parse MsgExec internal messages

### PR DESCRIPTION
**Pull request by ysong42**
Solution: fixed #525

### CODE REVIEW

File: usecase/parser/msg.go [hunk 1]
```diff
@@ -1294,22 +1295,67 @@ func ParseMsgExec(
...
+		parser := parserParams.ParserManager.GetParser(utils.CosmosParserKey(innerMsgType), utils.ParserBlockHeight(blockHeight))
+
+		msgCommands := parser(utils.CosmosParserParams{
+			AddressPrefix:   parserParams.AddressPrefix,
+			StakingDenom:    parserParams.StakingDenom,
+			TxsResult:       parserParams.TxsResult,
+			MsgCommonParams: parserParams.MsgCommonParams,
+			Msg:             innerMsg,
+			MsgIndex:        parserParams.MsgIndex,
+			ParserManager:   parserParams.ParserManager,
```
Comment (davcrypto) [inline usecase/parser/msg.go:1352]: why we need to pass `ParserManager` here?

**Review by davcrypto — COMMENTED** [review]
Thanks @ysong42. I left a comment for discussion. Other than that LGTM.

File: usecase/parser/msg.go (hunk 1 above)
Comment (ysong42) [reply usecase/parser/msg.go:1352]: This is mainly for parsing inner msgs for MsgExec. After parsing MsgExec, we will know its inner msg type, and then will need this `ParserManager` to retrieve the corresponding parser for parsing the inner msg.

Comment (davcrypto) [reply usecase/parser/msg.go:1352]: would the inner messages need `ParserManager`?

File: usecase/parser/msg_exec_test.go [hunk 2]
```diff
@@ -78,6 +97,22 @@ var _ = Describe("ParseMsgCommands", func() {
...
+
+			Expect(json.MustMarshalToString(createMsgSendEvent)).To(Equal(
+				strings.Replace(
+					regex.ReplaceAllString(expectedInnerMsg, ""),
+					"{UUID}",
+					createMsgSendEvent.UUID(),
+					-1,
+				),
+			))
+
```
Comment (calvinaco) [inline usecase/parser/msg_exec_test.go:115]: Can we have 1-2 more examples of MsgExec with a more complicated inner message? Want to make sure the parser is correct for diff message types.

Comment (ysong42) [reply usecase/parser/msg_exec_test.go:115]: Sure! Let me check if I could find more MsgExec related tx.

File: usecase/parser/msg.go (hunk 1 above)
Comment (ysong42) [reply usecase/parser/msg.go:1352]: @davcrypto I am so sorry... I missed the message 🙏  I will always refresh the page later on...

Yes, the inner messages will need the `ParserManager`. As the `ParseMsgExec()` could only parsed out a `model.RawMsgExec`, which does not handle the inner message.

We need to find the inner message type first, and then get the parser function from `ParserManager`.

**Review by davcrypto — APPROVED** [review]

File: usecase/parser/msg_exec_test.go (hunk 2 above)
Comment (ysong42) [reply usecase/parser/msg_exec_test.go:115]: @calvinaco Hi, I just added one more MsgExec example, whose inner msg is `MsgDelegate`.

**Pull request merged**